### PR TITLE
add FIREBASE_DATABASE_EMULATOR_HOST_VAR

### PR DESCRIPTION
**Pull request by IanWyszynski**
Support specifying database endpoint with FIREBASE_DATABASE_EMULATOR_HOST.

### CODE REVIEW

File: test/unit/firebase-app.spec.ts [hunk 1]
```diff
@@ -433,7 +433,8 @@ describe('FirebaseApp', () => {
       return app2.delete();
     });
 
-    it('should throw when databaseURL is not set', () => {
+    it('should throw when databaseURL is not set, will fail if ' +
+      'FIREBASE_DATABASE_EMULATOR_HOST is set', () => {
```
Comment (hiranya911) [inline test/unit/firebase-app.spec.ts]: Let's unset the env variable in the test.

```
delete process.env['FIREBASE_DATABASE_EMULATOR_HOST'];
```

File: test/unit/firebase.spec.ts [hunk 2]
```diff
@@ -167,7 +167,8 @@ describe('Firebase', () => {
       }).to.throw('The default Firebase app does not exist.');
     });
 
-    it('should throw given no databaseURL key when initializing the app', () => {
+    it('should throw given no databaseURL key when initializing the app,' +
+      ' will fail if FIREBASE_DATABASE_EMULATOR_HOST is set', () => {
```
Comment (hiranya911) [inline test/unit/firebase.spec.ts]: Here too.

File: test/unit/firebase-app.spec.ts [hunk 3]
```diff
@@ -433,7 +433,8 @@ describe('FirebaseApp', () => {
       return app2.delete();
     });
 
-    it('should throw when databaseURL is not set', () => {
+    it('should throw when databaseURL is not set, will fail if ' +
```
Comment (hiranya911) [inline test/unit/firebase-app.spec.ts]: Can we add a test case for the new functionality? Set the env variable and see app.database() works as intended.

**Comment by IanWyszynski** [comment]
This is currently blocked on the minor version bump in https://github.com/firebase/firebase-js-sdk/pull/2005, which must get merged first.

File: package.json [hunk 4]
```diff
@@ -54,7 +54,7 @@
   ],
   "types": "./lib/index.d.ts",
   "dependencies": {
-    "@firebase/database": "^0.4.7",
+    "@firebase/database": "^0.4.8",
```
Comment (hiranya911) [inline package.json]: You will have to run `npm install` on your working copy and push the updated package lock file for the tests to pass.

Comment (IanWyszynski) [reply package.json]: Actually, the firebase-js-sdk won't be exposing the variable so we won't need to change this at all.

File: src/database/database.ts [hunk 5]
```diff
@@ -5,6 +5,14 @@ import {Database} from '@firebase/database';
 
 import * as validator from '../utils/validator';
 
+/**
+ * This variable is re-defined in the firebase-js-sdk. Before modifying this
```
Comment (egilmorez) [inline src/database/database.ts]: "redefined" doesn't really need a hyphen.

**Pull request merged**